yb-voyager/cmd/importData.go

```diff
@@ -445,12 +463,45 @@ func importData(importFileTasks []*ImportFileTask) {
 				progressReporter.ImportFileStarted(task, totalProgressAmount)
 				importedProgressAmount := getImportedProgressAmount(task, state)
 				progressReporter.AddProgressAmount(task, importedProgressAmount)
+
+				var currentProgress int64
 				updateProgressFn := func(progressAmount int64) {
+					currentProgress += progressAmount
 					progressReporter.AddProgressAmount(task, progressAmount)
 				}
```

Comment:
You won't need the subsequent go-routine if you change `updateProgressFn` as:
```
				currentProgress := importedProgressAmount
				updateProgressFn := func(progressAmount int64) {
					currentProgress += progressAmount
					progressReporter.AddProgressAmount(task, progressAmount)
					importDataTableMetrics, _ := createImportDataTableMetrics(task.TableName,
							currentProgress, totalProgressAmount, 1)
					controlPlane.UpdateImportedRowCount(
								[]*cp.SnapshotImportTableMetrics{&importDataTableMetrics})
				}
```
Inside the `controlPlane.UpdateImportedRowCount()` you should track the timestamp at which last event was published and use it to publish one event per 10 seconds.